File: app/models/department_summary.py
from app.extensions import db


class DepartmentSummary(db.Model):
    __tablename__ = "department_summaries"

    id = db.Column(db.Integer, primary_key=True)
    snapshot_id = db.Column(db.Integer, db.ForeignKey("snapshots.id", ondelete="CASCADE"), nullable=False)
    department = db.Column(db.String(50), nullable=False)
    total_gifts = db.Column(db.Integer)
    total_amount = db.Column(db.Numeric(12, 2))
    goal = db.Column(db.Numeric(12, 2))
    pct_to_goal = db.Column(db.Numeric(8, 6))
    # Legacy-specific
    avg_gift = db.Column(db.Numeric(12, 2))
    new_expectancies = db.Column(db.Integer)
    open_estates = db.Column(db.Integer)
    recorded_expectancies = db.Column(db.Integer)
    # Events-specific (Third Party)
    third_party_total_gifts = db.Column(db.Integer)
    third_party_total_amount = db.Column(db.Numeric(12, 2))
    third_party_goal = db.Column(db.Numeric(12, 2))
    third_party_pct_to_goal = db.Column(db.Numeric(8, 6))

    def __repr__(self):
        return f"<DepartmentSummary {self.department}>"

File: src/models/departmentSummary.js
const { DataTypes } = require('sequelize');

module.exports = (sequelize) => {
  return sequelize.define('DepartmentSummary', {
    id: { type: DataTypes.INTEGER, primaryKey: true, autoIncrement: true },
    snapshotId: { type: DataTypes.INTEGER, allowNull: false, field: 'snapshot_id' },
    department: { type: DataTypes.STRING(50), allowNull: false },
    totalGifts: { type: DataTypes.INTEGER, field: 'total_gifts' },
    totalAmount: { type: DataTypes.DECIMAL(12, 2), field: 'total_amount' },
    goal: { type: DataTypes.DECIMAL(12, 2) },
    pctToGoal: { type: DataTypes.DECIMAL(8, 6), field: 'pct_to_goal' },
    // Legacy-specific
    avgGift: { type: DataTypes.DECIMAL(12, 2), field: 'avg_gift' },
    newExpectancies: { type: DataTypes.INTEGER, field: 'new_expectancies' },
    openEstates: { type: DataTypes.INTEGER, field: 'open_estates' },
    recordedExpectancies: { type: DataTypes.INTEGER, field: 'recorded_expectancies' },
    // Events-specific (Third Party)
    thirdPartyTotalGifts: { type: DataTypes.INTEGER, field: 'third_party_total_gifts' },
    thirdPartyTotalAmount: { type: DataTypes.DECIMAL(12, 2), field: 'third_party_total_amount' },
    thirdPartyGoal: { type: DataTypes.DECIMAL(12, 2), field: 'third_party_goal' },
    thirdPartyPctToGoal: { type: DataTypes.DECIMAL(8, 6), field: 'third_party_pct_to_goal' },
  }, {
    tableName: 'department_summaries',
    timestamps: false,
  });
};
